src/utils/change_rundir.cpp

```diff
@@ -39,7 +42,13 @@ void ChangeRunDir(const char *pdir) {
 
   if (pdir == nullptr || *pdir == '\0') return;
 
-  mkdir(pdir, 0775);
+  if (!fs::exists(pdir)) {
+    if (!fs::create_directories(pdir)) {
+      msg << "Failed to create directory: " << pdir << std::endl;
+      PARTHENON_FAIL(msg);
```

Comment:
I know you copied the code below, but I think we want to throw an exception here if we can.
```suggestion
      PARTHENON_THROW(msg);
```